web/src/js/components/FlowView/Details.jsx

```diff
@@ -136,8 +139,12 @@ export default function Details({ flow }) {
             <h4>Client Connection</h4>
             <ConnectionInfo conn={flow.client_conn}/>
 
-            <h4>Server Connection</h4>
-            <ConnectionInfo conn={flow.server_conn}/>
+            {flow.server_conn.address && [
+                    <div>
+                    <h4>Server Connection</h4>
+                    <ConnectionInfo conn={flow.server_conn}/>
+                    </div>
+            ]}
```

Comment:
Nit: We can use [`React.Fragment`](https://reactjs.org/docs/fragments.html) and don't need a new div. :)